backend/cfg/cfgize.ml

```diff
@@ -630,7 +630,8 @@ module Stack_offset_and_exn = struct
    fun ~stack_offset ~traps term ->
     check_and_set_stack_offset term ~stack_offset ~traps;
     match term.desc with
-    | Tailcall_self _ when List.length traps <> 0 || stack_offset <> 0 ->
+    | Tailcall_self _
+      when stack_offset <> 0 || match traps with _ :: _ -> true | [] -> false ->
```

Comment:
Please factor out the `match  traps ... ` part into a separate function `is_empty_list`, it will make it much easier to understand the condition. A alternative is to use `List.compare_length_with traps 0 <> 0`.